libcutefish build failed: outputmodel.cpp:606:21: error: ‘class KScreen::Output’ has no member named ‘setLogicalSize’

**Slackware 15 / GCC 11.2.0**
**KDE Plasma 5.23.4**


```
[ 62%] Building CXX object screen/CMakeFiles/cutefishscreen_qmlplugins.dir/outputmodel.cpp.o
/home/lancs/git/cutefishde-for-slackware/test/libcutefish/screen/outputmodel.cpp: In member function ‘bool OutputModel::setReplicationSourceIndex(int, int)’:
/home/lancs/git/cutefishde-for-slackware/test/libcutefish/screen/outputmodel.cpp:606:21: error: ‘class KScreen::Output’ has no member named ‘setLogicalSize’
  606 |         output.ptr->setLogicalSize(QSizeF());
      |                     ^~~~~~~~~~~~~~
/home/lancs/git/cutefishde-for-slackware/test/libcutefish/screen/outputmodel.cpp:617:21: error: ‘class KScreen::Output’ has no member named ‘setLogicalSize’
  617 |         output.ptr->setLogicalSize(source->logicalSize());
      |                     ^~~~~~~~~~~~~~
/home/lancs/git/cutefishde-for-slackware/test/libcutefish/screen/outputmodel.cpp:617:44: error: ‘class KScreen::Output’ has no member named ‘logicalSize’
  617 |         output.ptr->setLogicalSize(source->logicalSize());
      |                                            ^~~~~~~~~~~
make[2]: *** [screen/CMakeFiles/cutefishscreen_qmlplugins.dir/build.make:160: screen/CMakeFiles/cutefishscreen_qmlplugins.dir/outputmodel.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:421: screen/CMakeFiles/cutefishscreen_qmlplugins.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

**full build log**

```
[  1%] Automatic MOC and UIC for target cutefishaccounts_qmlplugins
[  1%] Built target cutefishaccounts_qmlplugins_autogen
[  2%] Generating user_interface.cpp, user_interface.h
[  2%] Generating user_interface.moc
[  3%] Generating accounts_interface.cpp, accounts_interface.h
[  4%] Generating accounts_interface.moc
[  5%] Building CXX object accounts/CMakeFiles/cutefishaccounts_qmlplugins.dir/cutefishaccounts_qmlplugins_autogen/mocs_compilation.cpp.o
[  6%] Building CXX object accounts/CMakeFiles/cutefishaccounts_qmlplugins.dir/accountsmanager.cpp.o
[  7%] Building CXX object accounts/CMakeFiles/cutefishaccounts_qmlplugins.dir/useraccount.cpp.o
[  8%] Building CXX object accounts/CMakeFiles/cutefishaccounts_qmlplugins.dir/usersmodel.cpp.o
[  9%] Building CXX object accounts/CMakeFiles/cutefishaccounts_qmlplugins.dir/qmlplugins.cpp.o
[ 10%] Building CXX object accounts/CMakeFiles/cutefishaccounts_qmlplugins.dir/user_interface.cpp.o
[ 11%] Building CXX object accounts/CMakeFiles/cutefishaccounts_qmlplugins.dir/accounts_interface.cpp.o
[ 12%] Linking CXX shared library libcutefishaccounts_qmlplugins.so
[ 12%] Built target cutefishaccounts_qmlplugins
[ 13%] Automatic MOC and UIC for target cutefishbluez_qmlplugins
[ 13%] Built target cutefishbluez_qmlplugins_autogen
[ 14%] Building CXX object bluez/CMakeFiles/cutefishbluez_qmlplugins.dir/cutefishbluez_qmlplugins_autogen/mocs_compilation.cpp.o
[ 15%] Building CXX object bluez/CMakeFiles/cutefishbluez_qmlplugins.dir/declarativemanager.cpp.o
[ 16%] Building CXX object bluez/CMakeFiles/cutefishbluez_qmlplugins.dir/declarativeadapter.cpp.o
[ 17%] Building CXX object bluez/CMakeFiles/cutefishbluez_qmlplugins.dir/declarativedevice.cpp.o
[ 18%] Building CXX object bluez/CMakeFiles/cutefishbluez_qmlplugins.dir/declarativeinput.cpp.o
[ 19%] Building CXX object bluez/CMakeFiles/cutefishbluez_qmlplugins.dir/declarativemediaplayer.cpp.o
[ 20%] Building CXX object bluez/CMakeFiles/cutefishbluez_qmlplugins.dir/declarativedevicesmodel.cpp.o
[ 21%] Building CXX object bluez/CMakeFiles/cutefishbluez_qmlplugins.dir/bluezqtextensionplugin.cpp.o
[ 21%] Building CXX object bluez/CMakeFiles/cutefishbluez_qmlplugins.dir/applet/devicesproxymodel.cpp.o
[ 22%] Building CXX object bluez/CMakeFiles/cutefishbluez_qmlplugins.dir/applet/bluetoothagent.cpp.o
[ 23%] Building CXX object bluez/CMakeFiles/cutefishbluez_qmlplugins.dir/applet/bluetoothmanager.cpp.o
[ 24%] Linking CXX shared library libcutefishbluez_qmlplugins.so
[ 24%] Built target cutefishbluez_qmlplugins
[ 25%] Automatic MOC and UIC for target cutefishmpris_qmlplugins
[ 25%] Built target cutefishmpris_qmlplugins_autogen
[ 26%] Building CXX object mpris/CMakeFiles/cutefishmpris_qmlplugins.dir/cutefishmpris_qmlplugins_autogen/mocs_compilation.cpp.o
[ 27%] Building CXX object mpris/CMakeFiles/cutefishmpris_qmlplugins.dir/mpris.cpp.o
[ 28%] Building CXX object mpris/CMakeFiles/cutefishmpris_qmlplugins.dir/mpriscontroller.cpp.o
[ 29%] Building CXX object mpris/CMakeFiles/cutefishmpris_qmlplugins.dir/mprismanager.cpp.o
[ 30%] Building CXX object mpris/CMakeFiles/cutefishmpris_qmlplugins.dir/mprisplayer.cpp.o
[ 31%] Building CXX object mpris/CMakeFiles/cutefishmpris_qmlplugins.dir/mprisplayeradaptor.cpp.o
[ 32%] Building CXX object mpris/CMakeFiles/cutefishmpris_qmlplugins.dir/mprisplayerinterface.cpp.o
[ 33%] Building CXX object mpris/CMakeFiles/cutefishmpris_qmlplugins.dir/mprisrootadaptor.cpp.o
[ 34%] Building CXX object mpris/CMakeFiles/cutefishmpris_qmlplugins.dir/mprisrootinterface.cpp.o
[ 35%] Building CXX object mpris/CMakeFiles/cutefishmpris_qmlplugins.dir/qmlplugins.cpp.o
[ 36%] Building CXX object mpris/CMakeFiles/cutefishmpris_qmlplugins.dir/qtdbusextended/dbusextendedabstractinterface.cpp.o
[ 37%] Building CXX object mpris/CMakeFiles/cutefishmpris_qmlplugins.dir/qtdbusextended/dbusextendedpendingcallwatcher.cpp.o
[ 38%] Linking CXX shared library libcutefishmpris_qmlplugins.so
[ 38%] Built target cutefishmpris_qmlplugins
[ 39%] Automatic MOC and UIC for target cutefishnm_qmlplugins
[ 39%] Built target cutefishnm_qmlplugins_autogen
[ 40%] Building CXX object networkmanagement/CMakeFiles/cutefishnm_qmlplugins.dir/cutefishnm_qmlplugins_autogen/mocs_compilation.cpp.o
[ 41%] Building CXX object networkmanagement/CMakeFiles/cutefishnm_qmlplugins.dir/appletproxymodel.cpp.o
[ 42%] Building CXX object networkmanagement/CMakeFiles/cutefishnm_qmlplugins.dir/activeconnection.cpp.o
[ 43%] Building CXX object networkmanagement/CMakeFiles/cutefishnm_qmlplugins.dir/networkitemslist.cpp.o
[ 44%] Building CXX object networkmanagement/CMakeFiles/cutefishnm_qmlplugins.dir/networkmodel.cpp.o
[ 45%] Building CXX object networkmanagement/CMakeFiles/cutefishnm_qmlplugins.dir/networkmodelitem.cpp.o
[ 46%] Building CXX object networkmanagement/CMakeFiles/cutefishnm_qmlplugins.dir/wiressitemsettings.cpp.o
[ 46%] Building CXX object networkmanagement/CMakeFiles/cutefishnm_qmlplugins.dir/uiutils.cpp.o
[ 47%] Building CXX object networkmanagement/CMakeFiles/cutefishnm_qmlplugins.dir/identitymodel.cpp.o
[ 48%] Building CXX object networkmanagement/CMakeFiles/cutefishnm_qmlplugins.dir/handler.cpp.o
[ 49%] Building CXX object networkmanagement/CMakeFiles/cutefishnm_qmlplugins.dir/configuration.cpp.o
[ 50%] Building CXX object networkmanagement/CMakeFiles/cutefishnm_qmlplugins.dir/enabledconnections.cpp.o
[ 51%] Building CXX object networkmanagement/CMakeFiles/cutefishnm_qmlplugins.dir/enums.cpp.o
[ 52%] Building CXX object networkmanagement/CMakeFiles/cutefishnm_qmlplugins.dir/wifisettings.cpp.o
[ 53%] Building CXX object networkmanagement/CMakeFiles/cutefishnm_qmlplugins.dir/qmlplugins.cpp.o
[ 54%] Linking CXX shared library libcutefishnm_qmlplugins.so
[ 54%] Built target cutefishnm_qmlplugins
[ 55%] Automatic MOC and UIC for target cutefishscreen_qmlplugins
[ 55%] Built target cutefishscreen_qmlplugins_autogen
[ 56%] Building CXX object screen/CMakeFiles/cutefishscreen_qmlplugins.dir/cutefishscreen_qmlplugins_autogen/mocs_compilation.cpp.o
[ 57%] Building CXX object screen/CMakeFiles/cutefishscreen_qmlplugins.dir/common/control.cpp.o
[ 58%] Building CXX object screen/CMakeFiles/cutefishscreen_qmlplugins.dir/common/globals.cpp.o
[ 59%] Building CXX object screen/CMakeFiles/cutefishscreen_qmlplugins.dir/common/orientation_sensor.cpp.o
[ 60%] Building CXX object screen/CMakeFiles/cutefishscreen_qmlplugins.dir/common/utils.cpp.o
[ 61%] Building CXX object screen/CMakeFiles/cutefishscreen_qmlplugins.dir/confighandler.cpp.o
[ 62%] Building CXX object screen/CMakeFiles/cutefishscreen_qmlplugins.dir/outputmodel.cpp.o
/home/lancs/git/cutefishde-for-slackware/test/libcutefish/screen/outputmodel.cpp: In member function ‘bool OutputModel::setReplicationSourceIndex(int, int)’:
/home/lancs/git/cutefishde-for-slackware/test/libcutefish/screen/outputmodel.cpp:606:21: error: ‘class KScreen::Output’ has no member named ‘setLogicalSize’
  606 |         output.ptr->setLogicalSize(QSizeF());
      |                     ^~~~~~~~~~~~~~
/home/lancs/git/cutefishde-for-slackware/test/libcutefish/screen/outputmodel.cpp:617:21: error: ‘class KScreen::Output’ has no member named ‘setLogicalSize’
  617 |         output.ptr->setLogicalSize(source->logicalSize());
      |                     ^~~~~~~~~~~~~~
/home/lancs/git/cutefishde-for-slackware/test/libcutefish/screen/outputmodel.cpp:617:44: error: ‘class KScreen::Output’ has no member named ‘logicalSize’
  617 |         output.ptr->setLogicalSize(source->logicalSize());
      |                                            ^~~~~~~~~~~
make[2]: *** [screen/CMakeFiles/cutefishscreen_qmlplugins.dir/build.make:160: screen/CMakeFiles/cutefishscreen_qmlplugins.dir/outputmodel.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:421: screen/CMakeFiles/cutefishscreen_qmlplugins.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

--- a/screen/outputmodel.cpp
+++ b/screen/outputmodel.cpp
@@ -603,7 +603,7 @@ bool OutputModel::setReplicationSourceIndex(int outputIndex, int sourceIndex)
             return false;
         }
         m_config->setReplicationSource(output.ptr, nullptr);
-        output.ptr->setLogicalSize(QSizeF());
+        output.ptr->setExplicitLogicalSize(QSizeF());
         resetPosition(output);
     } else {
         const auto source = m_outputs[sourceIndex].ptr;
@@ -614,7 +614,7 @@ bool OutputModel::setReplicationSourceIndex(int outputIndex, int sourceIndex)
         m_config->setReplicationSource(output.ptr, source);
         output.posReset = output.ptr->pos();
         output.ptr->setPos(source->pos());
-        output.ptr->setLogicalSize(source->logicalSize());
+        output.ptr->setExplicitLogicalSize(source->logicalSize());
     }
 
     reposition();

--- a/screen/outputmodel.cpp
+++ b/screen/outputmodel.cpp
@@ -614,7 +614,7 @@ bool OutputModel::setReplicationSourceIndex(int outputIndex, int sourceIndex)
         m_config->setReplicationSource(output.ptr, source);
         output.posReset = output.ptr->pos();
         output.ptr->setPos(source->pos());
-        output.ptr->setExplicitLogicalSize(source->logicalSize());
+        output.ptr->setExplicitLogicalSize(source->explicitLogicalSize());
     }
 
     reposition();

